migrations/2018-08-20-00-migrate-to-create-member-and-verify.ts

```diff
@@ -0,0 +1,217 @@
+  return `https://storage.googleapis.com/raha-video/${memberId}/invite.mp4`;
+}
+
+async function migrateToCreateMemberAndVerify(db: Firestore, dryRun: boolean) {
+  const operationsCollection = db.collection("operations");
+  const operations = (await operationsCollection.orderBy("created_at").get())
+    .docs;
+
+  const invitedBy: { [memberId: string]: string } = {};
+
+  let createdOperations = 0;
+  let updatedOperations = 0;
+
+  const newGenesisVerifyOperations: string[] = [];
+
+  for (const i in operations) {
+    const operation = operations[i];
+    const operationData = operation.data() as Operation;
+
+    if (operationData.op_code === OperationType.CREATE_MEMBER) {
+      if (operationData.data.request_invite_from_member_id) {
+        invitedBy[operationData.creator_uid] =
+          operationData.data.request_invite_from_member_id;
+      }
+    }
+
+    // Skip non-request-invite and non-trust operations
+    if (
+      operationData.op_code !== OperationType.REQUEST_INVITE &&
+      operationData.op_code !== OperationType.TRUST
+    ) {
+      continue;
+    }
+
+    if (operationData.op_code === OperationType.REQUEST_INVITE) {
+      const newMemberMemberId = operationData.creator_uid;
+      const operationToUid = operationData.data.to_uid;
+
+      const inviterMemberId =
+        operationToUid === null &&
+        GENESIS_REQUEST_INVITE_OPS.indexOf(operation.id) >= 0
+          ? GENESIS
+          : operationToUid;
+
+      if (invitedBy[newMemberMemberId] !== undefined) {
+        console.error("Member has already requested an invite!");
+      } else {
+        invitedBy[newMemberMemberId] = inviterMemberId;
+      }
+
+      const updatesToOperation = {
+        op_code: OperationType.CREATE_MEMBER,
+        "data.request_invite_from_member_id": inviterMemberId,
+        "data.to_uid": admin.firestore.FieldValue.delete()
+      };
+      console.log("=============================");
+      console.log("Updating operation", operation.id);
+      console.log("Operation was", operationData);
+      console.log("Update", updatesToOperation);
+      console.log("=============================");
+      if (!dryRun) {
+        await operation.ref.update(updatesToOperation);
+        ++updatedOperations;
+      }
+    }
+    if (operationData.op_code === OperationType.TRUST) {
+      const opCreatorUid = operationData.creator_uid;
+
+      const trusterMemberId =
+        opCreatorUid === null &&
+        ORIGINAL_GENESIS_TRUST_OPS.indexOf(operation.id) >= 0
+          ? GENESIS
+          : opCreatorUid;
+      const trustedMemberId = operationData.data.to_uid;
+
+      if (invitedBy[trustedMemberId] === trusterMemberId) {
+        // Check if this is a request-invite confirming Trust operation
+        // Check if the requisite "Verify" operation already exists.
+        // Allows this migration to be idempotent.
+        if (
+          !(await operationsCollection
```

Comment:
just  want to confirm that querying as we're editing the collection is a safe thing to do. might be better to just keep in memory which verify operations we've created and then flush them at once.